paf-mvp-core-js/src/validation/io-consistency.validator.ts

```diff
@@ -0,0 +1,24 @@
+import { MessageVerificationResult } from '@core/crypto/verifier';
+import { MessageBase } from '@core/model';
+
+export class MessageConsistencyValidator {
+  constructor(public messageTTLinSec = 30) {}
+
+  verify(
+    message: MessageBase,
+    senderHost: string,
+    receiverHost: string,
+    timestampInSec: number
+  ): MessageVerificationResult {
+    const timeSpent = timestampInSec - message.timestamp;
+    const isUnderTTL = timeSpent < this.messageTTLinSec;
+    const hasSender = message.sender === senderHost;
+    const hasReceiver = message.receiver === receiverHost;
+    const isValid = isUnderTTL && hasReceiver && hasSender;
+    const errors = isValid ? [] : [new Error('Inconsistency in the message')];
```

Comment:
FAR we should consider throwing typed Exceptions (instead of strings) to refine error responses and (more important) logs and associated metrics